src/pudl_archiver/zenodo/api_client.py

```diff
@@ -352,14 +396,19 @@ async def upload(self, file: BinaryIO, filename: str):
         """
         params = {"access_token": self.upload_key}
         if self.new_deposition.links.bucket:
-            url = f"{self.new_deposition.links.bucket}/{filename}"
+            url = f"{self.new_deposition.links.bucket}"
         elif self.new_deposition.links.files:
-            url = f"{self.new_deposition.links.files}/{filename}"
+            url = f"{self.new_deposition.links.files}"
         else:
             raise RuntimeError("No file or bucket link available for deposition.")
 
-        logger.info(f"Uploading file {filename} to zenodo deposition.")
-        await self.session.put(url, params=params, data=file)
+        logger.info(f"POSTing file {filename} to Zenodo url {url}.")
+        raw_json = await self._check_resp(
```

Comment:
This was confusingly silently succeeding before - the [`PUT`](https://developers.zenodo.org/?shell#update27) is on a specific file and renames the file only. the [`POST`](https://developers.zenodo.org/?shell#create24) is to the files/bucket and creates a new file.

This was causing the `publish` to fail because the new deposition was completely empty.

Reply:
This is definitely strange, because `publish` worked for me using the `PUT`